### PR TITLE
Changes And Makes Shuttle Authorization Repeal Message Work Properly  

### LINKED ISSUE #23123: "All authorizations to launch the shuttle early have been revoked" shows up when it shouldn't
This has been an issue for a while, sometimes it shows up, sometimes it doesn't, and it's never the right place. For instance this, in a span of a few seconds:

Attention:
The emergency shuttle will launch in 10 seconds

Attention:
All authorizations to launch the shuttle early have been revoked.

Priority Announcement

The Emergency Shuttle has left the station. Estimate 2 minutes until the shuttle docks at Central Command.

### PR DESCRIPTION
**Pull request by JohnTheDonut**
## **Changes Message And Fixes Shuttle Authorization Repeal Announcement**
Fixes shuttle repeal message so it actually works also changes shuttle repeal message to "Authorization repeal detected [remaining] authorizations needed" This means the shuttle will send a alert anytime an Authorization repeal is detected on the shuttle computer this should also help crew understand the status of the shuttle and if its going to blast off land leave them behind. 

## **Changelog**

:cl: Tofa01
Tweak: Changed alert message on early launch Authorization shuttle repeal message.
Fix: Makes the repeal message work and push a alert to the crew properly, also reports every Authorization repeal now.
/:cl:

### CODE REVIEW

File: code/modules/shuttle/emergency.dm [hunk 1]
```diff
@@ -86,13 +86,14 @@
...
 		var/remaining = auth_need - authorized.len
 		if(authorized.len && remaining)
 			minor_announce("[remaining] authorizations \
 				needed until shuttle is launched early", null, alert)
-		else
-			minor_announce("All authorizations to launch the shuttle \
-				early have been revoked.")
+		if(repeal)
+			minor_announce("Authorization repeal detected \
+				[remaining] authorizations needed")
```
Comment (coiax) [inline code/modules/shuttle/emergency.dm]: I don't like this phrasing.

Comment (Tofa01) [inline code/modules/shuttle/emergency.dm]: Of the message or the code itself?

Comment (coiax) [inline code/modules/shuttle/emergency.dm]: The message.

Comment (Tofa01) [inline code/modules/shuttle/emergency.dm]: Hm, what would you suggest? Looking back at it I at how it may confuse players.

Comment (Tofa01) [inline code/modules/shuttle/emergency.dm]: How about "Early Launch Authorizationon revoked [remaining] authorizations needed"?

Comment (lzimann) [inline code/modules/shuttle/emergency.dm]: also no multilines on this!

File: code/modules/shuttle/emergency.dm [hunk 2]
```diff
@@ -86,13 +86,13 @@
...
+		var/repeal = (authorized.len < old_len)
 		var/remaining = auth_need - authorized.len
 		if(authorized.len && remaining)
 			minor_announce("[remaining] authorizations \
 				needed until shuttle is launched early", null, alert)
-		else
-			minor_announce("All authorizations to launch the shuttle \
-				early have been revoked.")
+		if(repeal)
+			minor_announce("Early launch authorization revoked [remaining] authorizations needed")
```
Comment (ChangelingRain) [inline code/modules/shuttle/emergency.dm]: "Early launch authorization revoked. [remaining] authorizations needed."

Comment (coiax) [inline code/modules/shuttle/emergency.dm]: Early launch authorization revoked, [remaining] authorizations needed

Put a comma in.

File: code/modules/shuttle/emergency.dm [hunk 3]
```diff
@@ -86,13 +86,13 @@
 
 	if((old_len != authorized.len) && !ENGINES_STARTED)
 		var/alert = (authorized.len > old_len)
+		var/repeal = (authorized.len < old_len)
 		var/remaining = auth_need - authorized.len
 		if(authorized.len && remaining)
 			minor_announce("[remaining] authorizations \
```
Comment (ChangelingRain) [inline code/modules/shuttle/emergency.dm]: Also add a period/de-linebreak this one too.

**Comment by coiax** [comment]
@ChangelingRain jinx

File: code/modules/shuttle/emergency.dm (hunk 3 above)
Comment (Tofa01) [inline code/modules/shuttle/emergency.dm]: Do you mean add both vars to the same line like var/alert = (authorized.len > old_len), var/repeal = (authorized.len < old_len) ?

File: code/modules/shuttle/emergency.dm [hunk 4]
```diff
@@ -85,14 +85,13 @@
 				. = TRUE
 
 	if((old_len != authorized.len) && !ENGINES_STARTED)
-		var/alert = (authorized.len > old_len)
+		var/alert = (authorized.len > old_len), var/repeal = (authorized.len < old_len)
```
Comment (coiax) [inline code/modules/shuttle/emergency.dm]: Did @ChangelingRain say that this was okay? I do not approve

Comment (Tofa01) [inline code/modules/shuttle/emergency.dm]: Mixed requests == confused

Comment (ChangelingRain) [inline code/modules/shuttle/emergency.dm]: this is not okay OR what I requested

Comment (optimumtact) [inline code/modules/shuttle/emergency.dm]: What does this actually do

**Comment by Tofa01** [comment]
Ok if there is nothing else wrong with this it is ready to be merged whenever i guess.

**Pull request merged**